PowerPointLabs/PowerPointLabs/PictureSlidesLab/View/AdjustImageWindow.xaml.cs

```diff
@@ -256,5 +264,50 @@ private void ZoomOutButton_OnClick(object sender, RoutedEventArgs e)
         {
             _croppingAdorner.ZoomCroppingRect(-AdjustUnit);
         }
+
+        private void RotateFlipImg(RotateFlipType roatateFlipType)
+        {
+            var img = (Bitmap)Bitmap.FromFile(CropResult);
+            img.RotateFlip(roatateFlipType);
+            String rotatedImg = StoragePath.GetPath("crop-"
+                                    + DateTime.Now.GetHashCode() + "-"
+                                    + Guid.NewGuid().ToString().Substring(0, 7));
+            img.Save(rotatedImg);
+            thumbnailImageFile.Text = rotatedImg;
+            CropResult = rotatedImg;
+            CropResultThumbnail = rotatedImg;
+
+            Dispatcher.Invoke(DispatcherPriority.SystemIdle, new Action(() =>
```

Comment:
Does `Dispatcher.BeginInvoke` (async) work for this case? Better not use `Invoke` (sync) and with this priority